### ISSUE TITLE
Add option to `file` source to exclude older files

### ISSUE THREAD

**Issue by fuchsnj** [vectordotdev/vector#10005]
If you have a single log file that is written to, and rotated periodically (but kept in the same folder), it would be nice to allow Vector to not open a file-watcher on the older files. The number of files that are watched can easily exceed the file limit in this scenario.

The `ignore_older_secs` config option initially seems like it would solve this issue, but this only controls _where_ in a file Vector starts reading from. (If it's older than the setting, it just reads from the end so it only gets new logs). A new setting to completely exclude old files from being watched could be useful.

**Comment by vamshivishnu** [comment]
Supporting this issue. 

Just because when you are file watching log file like Test*.log file when we have too many files on directory like Test-1.log, Test-2.log ... a live file is always a increment then we will have too many Test-xxx.log files in the directory but in theory only 1 live log file. at this case all files are scanned for watch list and getting too many open file issue. Some how Vector need to have knowledge to understand live file only watch as old files have no activity for long time.

**Comment by c-hui** [comment]
Yeah, I need to close the file handler when a file is not active, avoid too many opened files limit.

**Comment by vamshivishnu** [comment]
Any update on this @c-hui

**Comment by c-hui** [comment]
谢谢，来信已收到！

**Comment by bootswithdefer** [comment]
This is impacting me too, I have a syslog server that has thousands of log files and vector keeps files open that it's not reading from.

**Comment by vamshivishnu** [comment]
@c-hui and @fuchsnj  any update ?

**Comment by Bingmang** [comment]
Meet the same problem, the description of `ignore_older_secs` is quite confusing.

**Comment by Bingmang** [comment]
Hi, any updates? @c-hui @fuchsnj

**Comment by c-hui** [comment]
No updates yet unfortunately. And there are already 1.9k open issues.
